mmv1/products/gkeonprem/BareMetalCluster.yaml

```diff
@@ -0,0 +1,1003 @@
+                      - !ruby/object:Api::Type::KeyValuePairs
+                        name: "featureGates"
+                        description: |
+                          A map of feature names to bools that enable or disable
+                          experimental features.
+                          This field modifies the built-in default values from
+                          "k8s.io/kubernetes/pkg/features/kube_features.go". Consider the
+                          documentation for the features you are enabling or disabling. While we
+                          encourage feature developers to make it possible to enable and disable
+                          features, some changes may require node reboots, and some features may
+                          require careful coordination to retroactively disable.
+                      - !ruby/object:Api::Type::Integer
+                        name: "podPidsLimit"
+                        description: |
+                          The maximum number of PIDs in any pod.
+                          Lowering this may prevent container processes from forking after the
+                          change.
+                          Defaults to -1.
+                        default_value: -1
+                      - !ruby/object:Api::Type::Integer
+                        name: "registryPullQps"
+                        description: |
+                          The limit of registry pulls per second.
+                          Setting this value to 0 means no limit.
+                          Updating this field may impact scalability by changing the amount of
+                          traffic produced by image pulls.
+                          Defaults to 5.
+                        default_value: 5
+                      - !ruby/object:Api::Type::Integer
+                        name: "registryBurst"
+                        description: |
+                          The maximum size of bursty pulls, temporarily allows pulls to burst to this
+                          number, while still not exceeding registry_pull_qps.
+                          The value must not be a negative number.
+                          Updating this field may impact scalability by changing the amount of
+                          traffic produced by image pulls.
+                          Defaults to 10.
+                        default_value: 10
+                      - !ruby/object:Api::Type::Boolean
+                        name: "serializeImagePullsDisabled"
+                        description: |
+                          Prevents the Kubelet from pulling multiple images at a time.
+                          We recommend *not* changing the default value on nodes that run docker
+                          daemon with version  < 1.9 or an Another Union File System (Aufs) storage
+                          backend. Issue https://github.com/kubernetes/kubernetes/issues/10959 has
+                          more details.
```

Comment:
tests

Reply:
The subfields "cpuManagerPolicy", "cpuCfsQuotaDisabled", "cpuCfsQuotaPeriod", "featureGates", "podPidsLimit" have been removed from the API surface recently. Removed from TF config in this PR and added rest of subfields in tests

Reply:
This makes me slightly nervous. How stable is this API? I worry about adding fields which may be unsupported by the API at a future point